packages/arm-core/src/components/DrawerOverlay/DrawerOverlay.tsx

```diff
@@ -0,0 +1,489 @@
+      return
+    }
+
+    const drawerPosition = getDrawerPositionFromState(state)
+    const { clientY } = event.targetTouches[0]
+
+    initialDrawerPositionRef.current = drawerPosition
+    initialDragPositionRef.current = clientY
+    lastDragPositionRef.current = clientY
+
+    updateDrawerPositionTransform()
+    setIsDragging(true)
+  }
+
+  function handleGestureMove(event: TouchEvent<HTMLDivElement>) {
+    if (lastDragPositionRef.current === null) {
+      return
+    }
+
+    // We need to use a ref here to prevent unnecessary re-renders.
+    lastDragPositionRef.current = event.targetTouches[0].clientY
+
+    updateDrawerPositionTransform()
+  }
+
+  function handleGestureEnd() {
+    if (lastDragPositionRef.current === null) {
+      return
+    }
+
+    const newState = getStateFromPosition()
+
+    initialDrawerPositionRef.current = null
+    initialDragPositionRef.current = null
+    lastDragPositionRef.current = null
+
+    updateDrawerPositionTransform(newState)
+    setIsDragging(false)
+
+    if (onStateChange && newState !== state) {
+      onStateChange(newState)
+    }
+  }
+
+  const drawerContainerStyle: CSSProperties = {}
+
+  if (isMobile(mode)) {
+    // The drawer needs to be completely visible in the open position, but without the controls that are locked to it in the viewport.
+    drawerContainerStyle.top = `-${lockedControlsHeight}px`
+  }
+
+  drawerContainerStyle.transform = getDrawerPositionTransform()
+
+  return (
+    <DrawerMapOverlay>
+      {otherControls.length > 0 && (
+        <OtherControlsContainer mode={mode}>
+          {otherControls.map((control) => (
+            <Fragment key={control.id}>{control.node}</Fragment>
+          ))}
+        </OtherControlsContainer>
+      )}
+      <DrawerContainer
+        ref={drawerContainerRef}
+        mode={mode}
+        style={drawerContainerStyle}
+        animate={!isDragging}
+      >
+        <ControlsContainer ref={lockedControlsRef} mode={mode}>
+          {lockedControls.map((control) => (
+            <Fragment key={control.id}>{control.node}</Fragment>
+          ))}
+        </ControlsContainer>
+        <Drawer
+          mode={mode}
+          onTouchStart={handleGestureStart}
+          onTouchMove={handleGestureMove}
+          onTouchEnd={handleGestureEnd}
+          onTouchCancel={handleGestureEnd}
+        >
+          <DrawerHandle
+            type="button"
+            variant="blank"
+            icon={
+              isDesktop(mode) ? (
+                <HandleIcon isOpen={state === DrawerState.Open} />
+              ) : null
+            }
+            size={32}
+            iconSize={20}
+            title="Open paneel"
+            onClick={handleToggle}
+          />
+          <DrawerContent ref={drawerContentRef}>
+            {Children.map(children, (child, index) => {
+              if (!isValidElement(child)) {
+                return child
+              }
+
+              return cloneElement(child, {
```

Comment:
Shouldn't this get a `key` prop?

Reply:
It's not needed in this case since `child` is an object containing a key that was passed into it. In this case React will automatically add the proper key for us.